src/lib/operators/import_binary.cpp

```diff
@@ -86,8 +91,30 @@ std::shared_ptr<const Table> ImportBinary::_on_execute() {
   std::shared_ptr<Table> table;
   ChunkID chunk_count;
   std::tie(table, chunk_count) = _read_header(file);
+  std::vector<std::shared_ptr<Chunk>> chunks;
+  chunks.reserve(chunk_count);
   for (ChunkID chunk_id{0}; chunk_id < chunk_count; ++chunk_id) {
-    table->emplace_chunk(_import_chunk(file, table));
+    chunks.emplace_back(_import_chunk(file, table));
+  }
+
+  try {
+    // check if partitioning information is available
+    auto partition_schema = _read_partitioning_header(file);
+    table->set_partitioning_and_clear(partition_schema);
```

Comment:
Why do we have to clear? The table should be empty if we want to import into it.

Reply:
The table contains chunks because the constructor by default creates chunks. Hence we have to clear it.